extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/options/TlsCertificateReloadUtils.java

```diff
@@ -0,0 +1,133 @@
+package io.quarkus.vertx.http.runtime.options;
+
+import static io.quarkus.vertx.http.runtime.options.HttpServerOptionsUtils.getFileContent;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.function.Function;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.vertx.http.runtime.HttpConfiguration;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.net.KeyStoreOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.SSLOptions;
+
+/**
+ * Utility class to handle TLS certificate reloading.
+ */
+public class TlsCertificateReloadUtils {
+
+    public static long handleCertificateReloading(Vertx vertx, HttpServer server,
+            HttpServerOptions options, HttpConfiguration configuration) {
+        // Validation
+        if (configuration.ssl.certificate.reloadPeriod.isEmpty()) {
+            return -1;
+        }
+        if (configuration.ssl.certificate.reloadPeriod.get().toMillis() < 1000) {
+            throw new IllegalArgumentException(
+                    "Unable to configure TLS reloading - The reload period cannot be less than 1 second");
+        }
+        if (options == null) {
+            throw new IllegalArgumentException("Unable to configure TLS reloading - The HTTP server options were not provided");
+        }
+        SSLOptions ssl = options.getSslOptions();
+        if (ssl == null) {
+            throw new IllegalArgumentException("Unable to configure TLS reloading - TLS/SSL is not enabled on the server");
+        }
+
+        Logger log = Logger.getLogger(TlsCertificateReloadUtils.class);
+        return vertx.setPeriodic(configuration.ssl.certificate.reloadPeriod.get().toMillis(), id -> {
```

Comment:
We might as well get rid of this lambda as well since all the rest of the code doesn't use them

Reply:
forgot that one :-D